provider/core/src/dynutil.rs

```diff
@@ -252,7 +252,7 @@ macro_rules! impl_dyn_provider {
 
          impl $crate::datagen::IterableDynProvider<$crate::serde::SerializeMarker> for $provider
          {
-             fn supported_options_for_key(&self, key: $crate::ResourceKey) -> Result<Box<dyn Iterator<Item = $crate::ResourceOptions> + '_>, $crate::DataError> {
+             fn supported_options_for_key(&self, key: $crate::ResourceKey) -> Result<std::boxed::Box<dyn Iterator<Item = $crate::ResourceOptions> + '_>, $crate::DataError> {
```

Comment:
Nit: `std` should be `alloc` (even, IMO, if it is in a context that requires std)

Reply:
`alloc` requires `extern crate alloc`. This is only used by datagen which is `std`